src/GeekLearning.Storage.Azure/AzureStore.cs

```diff
@@ -113,7 +78,7 @@ public AzureStore(string storeName, string connectionString, string containerNam
             blockBlob.Properties.ContentType = contentType;
             blockBlob.Properties.CacheControl = "max-age=300, must-revalidate";
             await blockBlob.SetPropertiesAsync();
-            return new Internal.AzureFileReference(blockBlob);
+            return new Internal.AzureFileReference(blockBlob, withMetadata: true);
```

Comment:
Yes, the `GetBlockBlobReference` method initializes a reference with an empty dictionary for metadata, and the properties are set just the line below